### ISSUE TITLE
Model class not found

### ISSUE THREAD

**Issue by timothyjeffcoat** [phpontrax/trax#3]
I created a classname extends ActiveRecord and the controller cannot recognize the class. the exception is "class not found" .

i verified that the class is not found with class_exists

What can be the issue here?

I did test with the console like
$u = new classname

and i did get back the table structure.

**Comment by westonwatson** [comment]
Just out of curiosity, where is the new ActiveRecord model/class located within your project? app/models ?

**Comment by timothyjeffcoat** [comment]
Correct.  

That is where I have them.

It seems like that some kind of refresh or reload needs to take place.  

The ActiveRecord classes are in the models directory but are not being recognized.

Note that the pre-existing classes I had in the models directory are working fine.  The problem is that the ones I recently added are not being recognized

**Comment by westonwatson** [comment]
As far as I can see, all classes in the models folder are loaded. We often just throw classes in there if we always want access to them (not just ActiveRecord models), sloppy- but you shouldn't have any trouble loading models from there, especially if their tables exist.

**Comment by timothyjeffcoat** [comment]
ok, my case is all lower.  I will go back and change that and respond with a report

**Comment by timothyjeffcoat** [comment]
I changed the ActiveRecord to camel case.  Still cannot be seen.

I created a new Class (not an ActiveRecord) , camel case, and placed it in the models directory. In the ide it can be seen but when placed on the CentOs server that class cannot be seen too.

Any ideas on what steps i might need to do to track this down?

**Comment by westonwatson** [comment]
I'm digging through the new source, we (in our environment) disabled a lot of cache features. Hopefully one of the current project devs will see this and comment/respond. If i find anything I'll comment back. Good Luck

**Comment by timothyjeffcoat** [comment]
Thanks for the help

**Comment by BugSmasher** [comment]
What is the filename?  It must be an underscored version of the ClassName (class_name.php)

**Comment by timothyjeffcoat** [comment]
It does not have an under score . let me try that out and i will report back

**Comment by timothyjeffcoat** [comment]
Just tried the Camel_Back naming for the file name as well as the class name and tested it on the server. 

The class is still not recognized.

Thanks anyways.

**Comment by BugSmasher** [comment]
Filename should be lowercase..  

app/models/class_name.php
--
<?php 

class ClassName extends ActiveRecord {

} ?>

**Comment by timothyjeffcoat** [comment]
ok. i tried the following

<?php

class CamelBack {

}

?>




the file name is

camel_back.php


the test code in the controller is:



                if (class_exists('CamelBack',true)) {
                    error_log("CamelBack is a Class!");
                    $camel = new CamelBack();
                }else{
                    error_log("CamelBack is not a class");
                }


The class is still not recognized.

**Comment by timothyjeffcoat** [comment]
I got it to work now.

I upgraded the vendor/trax to the current release.

So, my bad :-(

Thank you all for your help.

I was pleasantly surprised that anybody was using this framework.

From appearances I thought I was the only one , as well as the originator of the source code.

It is good to know that other people are using it.

**Comment by johnpipi** [comment]
Yes what they said is true.  Trax will take the class name if camelcase convert it to underscores to search for the filename.   I believe in the console its picky it is case sensitive so would have to be  $camel = new CamelBack()

Just our of curiosity what sites you guys using trax on?  I built gocoffeego.com all in trax.  But these days have been doing alot of rails development.

**Comment by johnpipi** [comment]
If you have any improvements would love to get a pull request :)
